### PR TITLE
Nuke regex

### PR DESCRIPTION
**Pull request by HaoK**
cc @blowdart @divega

Note you turn this feature off by setting the string to null or empty

### CODE REVIEW

File: src/Microsoft.AspNet.Identity/UserOptions.cs [hunk 1]
```diff
@@ -11,37 +11,12 @@ namespace Microsoft.AspNet.Identity
...
         /// </summary>
         /// <value>
-        /// The timeout value used after which user name validation via the <see cref="UserNameValidationRegex"/> will fail if it has not completed.
+        /// The list of allowed characters in the username used to validate user names.
         /// </value>
-        /// <remarks>
-        /// The default value is 20 milliseconds.
-        /// </remarks>
-        public TimeSpan UserNameValidationRegexTimeout { get; set; } = new TimeSpan(0,0,0,0,20);
+        public string AllowedUserNameCharacters { get; set; } = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@";
```
Comment (divega) [inline src/Microsoft.AspNet.Identity/UserOptions.cs]: I feel neglected that `áéíóúñÑüÜ` didn't make kit but if this is what we always had an no one complained, I guess it is ok :smile:

Comment (HaoK) [inline src/Microsoft.AspNet.Identity/UserOptions.cs]: I don't see any special characters in your username :). Or did it get englisized?

**Comment by divega** [comment]
:shipit:

**Comment by HaoK** [comment]
b2eb5feb8c26fc07f2794e32a94ebdd7cc805df5

**Pull request closed without merging**